src/quantuminspire/qiskit/circuit_parser.py

```diff
@@ -593,6 +599,23 @@ def _c_barrier(stream: StringIO, instruction: QasmQobjInstruction, binary_contro
 
         """
 
+    @staticmethod
+    def _delay(stream: StringIO, instruction: QasmQobjInstruction) -> None:
+        """ Translates the delay element for a qubit. In cQasm wait parameter is int. Actually only the default unit
+        "dt" will work correctly with cQasm, i.e. integer time unit depending on the target backend.
+
+        In qiskit/circuit/delay.py multiple units are defined.
+        In qiskit/circuit/instruction.py assemble() method looses the unit of the delay instruction. Only the
```

Comment:
That's the problem, SDK should abstract this away, but the problem is how because we don't 'hear' what unit was chosen by the circuit-programmer (check test_generate_cqasm_correct_output_delay_qarg where multiple units are chosen but we only hear the number). Also for calculating ms to hardware cycles we need to know what time a hardware cycle takes. So we 'assume' the circuit programmer used the default (is hardware cycles).